### PR TITLE
feat: support running cypress tests

### LINKED ISSUE #44: Add a flag to run the new cypress test as part of the /deploy tests
We use the `/ deploy` string to run the Selenium-based acceptance tests.
Not that we added [Cypress-based tests](https://github.com/SwissDataScienceCenter/renku/pull/2776), it would be nice to run them as well. We can initially make them optional by requiring a flag in the deploy string, something like `/ deploy #cypress`.
In the future, we can consider splitting cypress tests in multiple categories so that we don't run time consuming tests when they aren't needed `/ deeply #cypress=ui,notebooks,loadtests`

### PR DESCRIPTION
**Pull request by ciyer**
Support handling `#cypress` in the deploy command.

fix #44

### CODE REVIEW

File: test-renku-cypress/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Run Cypress-based acceptance tests
+
+This is a composite action that:
+- clones the renku repo and checks out the specified ref
+- runs the cypress acceptance tests from the renku repo
+- saves the results as an artifact
+
+# Interaction with Selenium-based acceptance tests
+
+The `test-rnku-cypress` action **does not** tear down the deployment, but the `test-renku` action to run the Selenium-based tests does, and some care needs to be taken when running the Cypress tests without running the Selenium tests.
```
Comment (andre-code) [inline test-renku-cypress/README.md:10]: typo in `test-rnku-cypress`

File: check-pr-description/action.yaml [hunk 2]
```diff
@@ -35,6 +35,9 @@ outputs:
   test-enabled:
     description: whether the tests should run or not
```
Comment (andre-code) [inline check-pr-description/action.yaml:36]: Could we clarity in this description that it only refers to Selenium tests?

**Review by andre-code — COMMENTED** [review]
This is going to be very helpful, thank you. I find it useful to clarify in the description the difference between selenium tests and cypress tests.

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
This looks great!
On top of the 2 comments from @andre-code , I was wondering if we need to keep the long commented sections on "We currently do not tear-down the renku deployment [...]" since you mentioned that on the Readme. I guess we can live with that for the moment being.
Would the alternative be moving the teardown somewhere else, so we execute them if any of the 2 tests run? Would you mind making a follow-up issue for that? (I fear I might miss something since I don't know much the actions)

**Comment by ciyer** [comment]
Thanks for the comments everyone. I have addressed the typo + clarification request. 

> Would the alternative be moving the teardown somewhere else, so we execute them if any of the 2 tests run? Would you mind making a follow-up issue for that? (I fear I might miss something since I don't know much the actions)

It depends on where we are headed. If the plan is to eventually switch to cypress and fade out the selenium tests, then the alternative would be to put the teardown in the cypress action (which is why I left those sections there, but commented out). If the plan is to continue to support both tests in parallel, then we would need to move the teardown to a separate step, as you suggest.

I can create the issue, but we should decide what the target is.

**Comment by olevski** [comment]
Just for completeness this is a PR where this new action is being used. https://github.com/SwissDataScienceCenter/renku-ui/pull/2182

**Review by olevski — APPROVED** [review]

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm! :rocket: 

> I can create the issue, but we should decide what the target is.

As discussed offline, let's add both options to the issue -- we can pick a solution in a couple of months after seeing how it goes with the Cypress tests.

**Pull request merged**